mern-ecommerce/controllers/auth.js

```diff
@@ -80,3 +80,13 @@ exports.isAdmin = (req, res, next) => {
   }
   next();
 };
+
+exports.checkUser = (req, res, next) => {
+  let user = req.profile && req.auth && req.profile._id == req.auth._id;
```

Comment:
This line of code is checking if the user is authenticated, but it's not very readable and can be improved. Consider breaking it down into multiple lines for better readability. Also, it's not clear what `req.profile` and `req.auth` are, consider adding a comment to explain what these variables represent.